### ISSUE TITLE
[Deploy] Android Messages

### ISSUE THREAD

**Issue by michaelsouellette** [meetfranz/plugins#164]
### Service
Android Messages

### Link to your repository
https://github.com/michaelsouellette/recipe-androidMessages

### Additional Information
Franz plugin for the new Android Messages interface

**Comment by efraimbart** [comment]
Code to enable the counter:

``` 
  const getMessages = function getMessages() {
    const elements = document.querySelectorAll('.yrs5ff:first-child');
    let count = 0;

    for (let i = 0; i < elements.length; i += 1) {
      if (elements[i].querySelectorAll('.Ht7Nrb').length === 0) {
        count += 1;
      }
    }

    // set Franz badge
    Franz.setBadge(count);
  };

**Comment by efraimbart** [comment]
I just realized that this was was originally not working because it was based off WhatsApp and not updated fully, in that case you should probably also remove the remaining WhatsApp css stuff as it's not applicable here.

**Comment by arjunkalburgi** [comment]
When will this go live? I've been trying to hack it to work with my Franz but haven't been successful. When will it appear in the "Add Services" search?

**Comment by michaelsouellette** [comment]
@askalburgi Can't comment on go live since it kind of depends on the main developer(s). However, you can pull this repo down and put it in the dev directory location for Franz and you should be able to use it as is. Make sure to check out the plugins repo for directions on how to do that. It has been working for me.

**Comment by michaelsouellette** [comment]
@efraimbart Yeah I was pretty hasty in creating this, I will take a look at removing this extra bit of code.

**Comment by michaelsouellette** [comment]
@efraimbart I removed the css it appears nothing is really needed for this plugin

**Comment by michaelsouellette** [comment]
If anyone has suggested improvements to this plugin feel free to open a pull request.

**Comment by bgschiller** [comment]
This is working great for me. I don't see anything to improve. 👍

**Comment by NobleMule** [comment]
I just tried knocking up an android messages recipe myself. Was going to proudly try to deploy it... and then saw this.

Anyway, not sure if it's 100% correct because I don't get tons of messages to be able to verify this but I think the presence of the `.yrs5ff` class is enough to signify a new message so maybe you could simplify your `getMessages` function by doing something like this:

```javascript
Franz.setBadge(document.querySelectorAll('.yrs5ff:first-child').length);
```

I've removed the loop in mine (although I did my query selector in a slightly less efficient way so this is untested) and it _appears_ to work...

**Comment by NobleMule** [comment]
Also, if you have no CSS, this can be omitted entirely:

```javascript
  // inject franz.css stylesheet
  Franz.injectCSS(path.join(__dirname, 'service.css'));
```

**Comment by efraimbart** [comment]
@NobleMule the `for` loop is to eliminate muted conversations, similar to WhatsApp. 
I've also created a pull request to entirely remove the CSS.

**Comment by NobleMule** [comment]
@efraimbart Ah, makes sense. Very nice 😄

**Comment by michaelsouellette** [comment]
@efraimbart I merged your pull request, thanks!

**Comment by mikezielonka** [comment]
Is this close to being deployed into a new version of the beta?  😇

**Comment by thatchrismiller** [comment]
Really hoping to see this soon. It's the only thing keeping me from rocking Franz right now.

**Comment by efraimbart** [comment]
@thatchrismiller It's quite easily to manually add this plugin to your Franz setup.

**Comment by bpwned** [comment]
@efraimbart Do you have instructions to manually add this handy? I'm having trouble finding them and I'd really like to try this.

**Comment by michaelsouellette** [comment]
@bpwned 
https://github.com/meetfranz/plugins/blob/master/docs/integration.md
Look under installation.

**Comment by FilipMalczak** [comment]
Duplicated by #192 and #170 . I personally like #192 approach best.

**Comment by megaroeny** [comment]
What's the progress on this? Where's the PR for this?
